exchange/utils.go

```diff
@@ -847,6 +854,60 @@ func mergeBidderRequests(allBidderRequests []BidderRequest, bidderNameToBidderRe
 	return allBidderRequests
 }
 
+func setLegacyGDPRFromGPP(r *openrtb2.BidRequest, gpp gpplib.GppContainer) {
+	if r.Regs != nil && r.Regs.GDPR == nil {
+		if r.Regs.GPPSID != nil {
+			// Set to 0 unless SID exists
+			r.Regs.GDPR = ptrutil.ToPtr[int8](0)
+			for _, id := range r.Regs.GPPSID {
+				if id == int8(gppConstants.SectionTCFEU2) {
+					regs := *r.Regs
```

Comment:
Please move the copy up a level so that it also applies to `r.Regs.GDPR = ptrutil.ToPtr[int8](0)`.

Reply:
Oh, yes. This is where I need the facepalm emoji. Thanks.